src/controllers/admin.ts

```diff
@@ -0,0 +1,66 @@
+import * as Hapi from 'hapi'
+import Config from '../services/Config'
+import PodDatabase from '../services/PodDatabase'
+import { Injector } from 'reduct'
+import { getCurrencyPerSecond } from '../util/priceRate'
+import BigNumber from 'bignumber.js'
+
+import { create as createLogger } from '../common/log'
+const log = createLogger('admin')
+
+export default function (server: Hapi.Server, deps: Injector) {
+  const podDatabase = deps(PodDatabase)
+  const config = deps(Config)
+
+  async function getAdminInfo (request: Hapi.Request, h: Hapi.ResponseToolkit) {
+    return {
+      test: 'This shows that you are running the Codiusd Admin API.'
+    }
+  }
+
+  async function getRunningPods (request: Hapi.Request, h: Hapi.ResponseToolkit) {
+    return {
+      running: podDatabase.getRunningPods()
+    }
+  }
+
+  async function getPodInfo (request: Hapi.Request, h: Hapi.ResponseToolkit) {
+    log.debug('Querying pod info for: ', request.query['id'])
+    return podDatabase.getPod(request.query['id'])
+  }
+
+  async function getAllUptime (request: Hapi.Request, h: Hapi.ResponseToolkit) {
+    const uptime = podDatabase.getLifetimePodsUptime()
+    const profit = uptime.times(getCurrencyPerSecond(config).div(new BigNumber(10).pow(config.hostAssetScale)))
```

Comment:
Might be worth adding a comment to note that if we ever add a feature to dynamically change the price then this calculation will no longer be accurate